xbmc/utils/DisplayInfo.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "DisplayInfo.h"
+
+#include "utils/log.h"
+
+extern "C"
+{
+#include <libdisplay-info/cta.h>
+#include <libdisplay-info/edid.h>
+#include <libdisplay-info/info.h>
+}
+
+using namespace KODI;
+using namespace UTILS;
+
+void CDisplayInfo::DiInfoDeleter::operator()(di_info* p)
+{
+  di_info_destroy(p);
+}
+
+std::unique_ptr<CDisplayInfo> CDisplayInfo::Create(std::vector<uint8_t> edid)
```

Comment:
static factory method allows us to only return a pointer if the edid is valid.

Reply:
We don't need a copy of the edid vector, so change to a const reference? Same for the `CDisplayInfo` constructor.